src/constants/diff-types.js

```diff
@@ -1,16 +1,37 @@
 export const diffTypes = {
-  HIGHLIGHTED_TEXT: 'highlighted text',
-  HIGHLIGHTED_SOURCE: 'highlighted source',
-  HIGLIGHTED_RENDERED: 'higlighted rendered',
-  SIDE_BY_SIDE_RENDERED: 'side-by-side rendered',
-  SIDE_BY_SIDE_TEXT: 'side-by-side text',
-};
-
-// TODO - map each diffType to it's corresponding endpoint string
-export const changeDiffTypes = {
-  [diffTypes.HIGHLIGHTED_TEXT]: 'html_text',
-  // Commented out because this differ has some issues right now
-  // [diffTypes.HIGHLIGHTED_SOURCE]: 'html_source',
-  [diffTypes.HIGHLIGHTED_SOURCE]: 'source',
-  [diffTypes.SIDE_BY_SIDE_RENDERED]: 'html_source'  // HACK
+  HIGHLIGHTED_TEXT: {
+    description: 'highlighted text',
+    diff_service: 'html_text',
+    value: 'HIGHLIGHTED_TEXT'
+  },
+  HIGHLIGHTED_SOURCE: {
+    description: 'highlighted source',
+    diff_service: 'html_source',
+    value: 'HIGHLIGHTED_SOURCE'
+  },
+  HIGHLIGHTED_RENDERED: {
+    description: 'highlighted rendered',
+    diff_service: 'TODO',
+    value: 'HIGHLIGHTED_RENDERED'
+  },
+  SIDE_BY_SIDE_RENDERED: {
+    description: 'side-by-side rendered',
+    diff_service: 'html_source',  // HACK, real side-by-side doesn't exist yet
+    value: 'SIDE_BY_SIDE_RENDERED'
+  },
+  SIDE_BY_SIDE_TEXT: {
+    description: 'side-by-side text',
+    diff_service: 'TODO',
+    value: 'SIDE_BY_SIDE_TEXT'
+  },
+  CHANGES_ONLY_TEXT: {
+    description: 'changes only text',
+    diff_service: 'html_text',
+    value: 'CHANGES_ONLY_TEXT'
+  },
+  CHANGES_ONLY_SOURCE: {
+    description :'changes only source',
+    diff_service: 'html_source',
+    value: 'CHANGES_ONLY_SOURCE'
+  }
```

Comment:
Do we really need `value` at all? I think we should get rid of it if we can. Otherwise, to alleviate future typos I know I will make, it might be good to do this:

```js
export const diffTypes = {
  HIGHLIGHTED_TEXT: {
    description: '',
    service: ''
    // no `value`
  },
  // etc...
};

for (let key in diffTypes) {
  diffTypes[key].value = key;
}
```

Reply:
Yea, this is kind of smelly. I'm using it to have the string value of the keys, which we then use in various places to get at the right object properties. There's probably a better way, but I can't figure out how without having to loop over the keys and compare them to hard-coded strings, which would be worse.

I guess we're trading in the previous approach's opaqueness with this approach's redundancy. This would be easier to resolve with a sit-down review so I could explain more easily and point out the different places removing it would affect.

Until then I used the `for` loop